src/master/filesystem_operations.cc

```diff
@@ -641,8 +641,11 @@ uint8_t fs_do_setlength(const FsContext &context, uint32_t inode, uint64_t lengt
 		return status;
 	}
 
-	fsnodes_setlength(static_cast<FSNodeFile*>(p), length);
-	fs_changelog(ts, "LENGTH(%" PRIu32 ",%" PRIu64 ")", inode, static_cast<FSNodeFile*>(p)->length);
+	bool eraseFurtherChunks = true;
```

Comment:
Consider to explain the reason of `eraseFurtherChunks = true`.
I think it's not trivial, specially because the reason does not rely in `fs_do_setlength()` function itself, but in the function which call `fs_do_setlength()`, like `matoclserv_chunk_status()` and `matoclserv_fuse_truncate`.

Reply:
Applied.